### PR TITLE
[CARBONDATA-173]The exception info is not proper when measure used as columnDict

### PR DESCRIPTION
**Pull request by Jay357089**
https://issues.apache.org/jira/browse/CARBONDATA-173

just change the error info and make it more proper, then add test case.

### CODE REVIEW

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 1]
```diff
@@ -489,9 +489,10 @@ object GlobalDictionaryUtil extends Logging {
     val preDictDimensionOption = dimensions.filter(
       _.getColName.equalsIgnoreCase(dimParent))
     if (preDictDimensionOption.length == 0) {
-      logError(s"No column $dimParent exists in ${table.getDatabaseName}.${table.getTableName}")
-      throw new DataLoadingException(s"No column $colName exists " +
-      s"in ${table.getDatabaseName}.${table.getTableName}")
+      logError(s"No key column $dimParent exists in ${table.getDatabaseName}.${table.getTableName}")
+      throw new DataLoadingException(s"No key column $colName exists in " +
+        s"${table.getDatabaseName}.${table.getTableName}, please make sure " +
+        s"that $colName can be dictionary.")
```
Comment (kumarvishal09) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: Please add a message only key column can be part of dictionary

Comment (Jay357089) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: done

**Comment by kumarvishal09** [comment]
LGTM

**Comment by Jay357089** [comment]
CI passed, pls check.
 http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/28/

**Pull request closed without merging**